apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -102,7 +62,15 @@ const LinkEditorModalContent = ({
     onSave(linkText, linkHref),
   )
 
-  const { siteId } = useQueryParse(editPageSchema)
+  // hacky solution to allow setting the correct state of "Add link" CTA button
+  // while not showing error message on initial render for better UX
+  const [isFirstRender, setIsFirstRender] = useState(true)
+  const showLinkTextErrorState: boolean =
+    !isFirstRender && !!errors.linkText?.message
```

Comment:
i think i might have fixed this in my pr (or at least for collections) - this triggers because we have a few required properties that are not set. instead of doing this, we can add an empty string though i have to note that this is technically not ideal (user can save w/ empty input)